google/auth/external_account.py

```diff
@@ -130,10 +130,7 @@ def __init__(
         self._default_scopes = default_scopes
         self._workforce_pool_user_project = workforce_pool_user_project
         self._universe_domain = universe_domain or credentials.DEFAULT_UNIVERSE_DOMAIN
-        self._trust_boundary = {
-            "locations": [],
-            "encoded_locations": "0x0",
-        }  # expose a placeholder trust boundary value.
+        self._trust_boundary = credentials.DEFAULT_TRUST_BOUNDARY
```

Comment:
This needs to a deep copy or trust boundary modifications would impact other credential objects.

Reply:
I put a deep copy at the `DEFAULT_TRUST_BOUNDARY`. Would that be OK?

Reply:
No, I think there is a misunderstanding. The code is hitting this error https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument. By creating a DEFAULT dict, the dict is shared by all credential objects, because in this line of code you are grabbing a pointer to that dict.

Deep copying once at the global namespace is a NOOP, a deep copy in the constructor would accomplish what I believe your intent for the default is.

My suggestion is to use `DEFAULT_TRUST_BOUNDARY` as a singleton if you want to share the cache across all credential types, otherwise codify the default dict in your constructor, to avoid a race condition.

Reply:
Oh, I made a copy on each use case.